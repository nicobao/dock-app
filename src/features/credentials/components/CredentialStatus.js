import {NBox, Typography} from '../../../design-system';
import {HStack} from 'native-base';
import React from 'react';
import {credentialStatusData} from '../credentials';
import {useGetCredentialStatus} from '@docknetwork/wallet-sdk-react-native/lib';

export function CredentialStatus({credential}) {
  const status = useGetCredentialStatus({credential});

  return (
    <HStack ml={3}>
      <NBox mt={0.5}>{credentialStatusData[status].icon}</NBox>
      <Typography
        color={credentialStatusData[status].color}
        ml={1}
        variant={'credentialIssuanceDate'}>
        {credentialStatusData[status].message}
      </Typography>
    </HStack>
  );
}
