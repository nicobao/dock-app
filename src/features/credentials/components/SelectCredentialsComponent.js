import {NBox, Typography} from '../../../design-system';
import {translate} from '../../../locales';
import {SingleCredentialToShare} from './SingleCredentialToShare';
import {EmptyCredentials} from '../CredentialsScreen';
import React from 'react';
import {addTestId} from '../../../core/automation-utils';

export function SelectCredentialsComponent({
  credentials,
  selectedCredentials,
  setSelectedCredentials,
}) {
  return (
    <NBox>
      <Typography variant="h1" mb={2}>
        {translate('credentials.select_what_to_share')}
      </Typography>

      {credentials.length > 0 ? (
        credentials.map((item, index) => (
          <SingleCredentialToShare
            {...addTestId(`credentials_${index}`)}
            key={item.id}
            rawCredential={item}
            checked={selectedCredentials[item.id]}
            onSelect={isSelected => {
              setSelectedCredentials(prev => {
                return {
                  ...prev,
                  [item.id]: isSelected ? item.content : undefined,
                };
              });
            }}
          />
        ))
      ) : (
        <EmptyCredentials mt={'50%'} />
      )}
    </NBox>
  );
}
