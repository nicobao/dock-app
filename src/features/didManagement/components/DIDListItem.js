import {
  Button,
  DotsVerticalIcon,
  IconButton,
  NBox,
  Theme,
  Typography,
} from '../../../design-system';
import {HStack, VStack} from 'native-base';
import {addTestId} from '../../../core/automation-utils';
import {translate} from '../../../locales';
import React from 'react';

export function DIDListItem({didDocumentResolution, onOptionClicked, onShare}) {
  const {didDocument} = didDocumentResolution;
  return (
    <NBox
      my={3}
      py={3}
      pl={4}
      pr={3}
      style={{
        backgroundColor: Theme.colors.cardItemBackground,
        borderRadius: 12,
      }}>
      <VStack>
        <HStack>
          <Typography
            style={{
              flexGrow: 1,
            }}
            numberOfLines={1}
            pt={2}
            variant="didTitle">
            {typeof didDocumentResolution.name === 'string' &&
            didDocumentResolution.name.length > 0
              ? didDocumentResolution.name
              : translate('didManagement.no_name_set')}
          </Typography>
          <IconButton
            col
            {...addTestId('DIDListItemOptionButton')}
            onPress={onOptionClicked}>
            <DotsVerticalIcon />
          </IconButton>
        </HStack>
        <HStack>
          <Typography numberOfLines={1} mb={7} pt={2} variant="didDescription">
            {didDocument.id}
          </Typography>
        </HStack>
        <Button
          width="30%"
          size="xs"
          variant={'whiteButton'}
          colorScheme="dark"
          {...addTestId('Share')}
          onPress={onShare}>
          <Typography color={Theme.colors.primaryBackground}>
            {translate('didManagement.share')}
          </Typography>
        </Button>
      </VStack>
    </NBox>
  );
}
