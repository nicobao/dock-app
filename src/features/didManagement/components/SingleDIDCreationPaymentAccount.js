import {NBox, Typography} from '../../../design-system';
import React from 'react';
import {HStack, Stack} from 'native-base';
import {TokenAmount} from '../../tokens/ConfirmTransactionModal';
export function SingleDIDCreationPaymentAccount({item}) {
  return (
    <NBox>
      <HStack mb={1}>
        <Typography
          style={{
            flexGrow: 1,
          }}
          numberOfLines={1}
          textAlign="left"
          variant="description">
          {item.label}
        </Typography>
        <TokenAmount amount={item.balance}>
          {({tokenAmount, tokenSymbol}) => (
            <>
              <Stack direction="column">
                <Typography variant="description">
                  {tokenAmount} {tokenSymbol}
                </Typography>
              </Stack>
            </>
          )}
        </TokenAmount>
      </HStack>
      <Typography
        numberOfLines={1}
        textAlign="left"
        variant="screen-description">
        {item.description}
      </Typography>
    </NBox>
  );
}
