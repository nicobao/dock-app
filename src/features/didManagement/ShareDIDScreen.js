import {Box, Stack} from 'native-base';
import React, {useCallback} from 'react';
import {Dimensions} from 'react-native';
import QRCode from 'react-native-qrcode-svg';
import {PolkadotIcon} from 'src/components/PolkadotIcon';
import {addTestId} from 'src/core/automation-utils';
import {Ionicons} from '@native-base/icons';
import {Icon} from 'native-base';
import {
  BackButton,
  Button,
  Content,
  Header,
  NBox,
  ScreenContainer,
  Theme,
  Typography,
} from '../../design-system';
import {translate} from '../../locales';
import Clipboard from '@react-native-community/clipboard';
import {showToast} from '../../core/toast';
import {ANALYTICS_EVENT, logAnalyticsEvent} from '../analytics/analytics-slice';
import {Image} from 'react-native';

export function ShareDIDScreen({did, didName, accountIcon, onCopyAddress}) {
  const qrSize = Dimensions.get('window').width * 0.5;

  return (
    <ScreenContainer {...addTestId('ShareDIDScreen')}>
      <Header>
        <BackButton {...addTestId('BackButton')} />
      </Header>
      <Stack alignItems="center">
        <Typography variant="h1" mb={2}>
          {translate('didManagement.share_did')}
        </Typography>
        <Typography variant="screen-description">
          {translate('didManagement.share_did_desc')}
        </Typography>
      </Stack>
      <Content marginLeft={26} marginRight={26}>
        <Stack
          backgroundColor={Theme.colors.cardItemBackground}
          p={5}
          alignItems="center"
          borderRadius={12}
          my={7}>
          <Stack p={2} bg={Theme.colors.qrCodeBackground}>
            <QRCode value={did} size={qrSize} />
          </Stack>
          <Stack direction="row" alignItems="center" flex={1} pt={5}>
            <Box pr={2}>
              <Image
                style={{
                  height: 40,
                  width: 40,
                }}
                source={require('../../assets/circle.png')}
              />
              {accountIcon}
            </Box>
            <Stack direction="column" alignItems="flex-start" flex={1}>
              <Typography variant="h3">{didName}</Typography>
              <Typography numberOfLines={1}>{did}</Typography>
            </Stack>
          </Stack>
        </Stack>
      </Content>
      <NBox mb={7} mx={7}>
        <Button
          startIcon={<Icon as={Ionicons} name="copy-outline" size="sm" />}
          size="sm"
          onPress={onCopyAddress}>
          {translate('didManagement.copy_did')}
        </Button>
      </NBox>
    </ScreenContainer>
  );
}

export function ShareDIDScreenContainer({route}) {
  const {did, didName} = route.params || {};

  const handleCopyAddress = useCallback(() => {
    Clipboard.setString(did);
    showToast({
      message: translate('didManagement.did_copied'),
    });
    logAnalyticsEvent(ANALYTICS_EVENT.DID.DID_SHARED, {
      did,
    });
  }, [did]);

  return (
    <ShareDIDScreen
      onCopyAddress={handleCopyAddress}
      did={did}
      didName={didName}
      accountIcon={<PolkadotIcon address={did} size={32} />}
    />
  );
}
