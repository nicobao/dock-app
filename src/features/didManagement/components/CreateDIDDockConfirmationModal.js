import {Modal} from '../../../components/Modal';
import React from 'react';
import {translate} from '../../../locales';
import {LoadingButton, NBox, Typography} from '../../../design-system';
import {VStack} from 'native-base';
import {addTestId} from '../../../core/automation-utils';
import {AmountDetails} from '../../tokens/ConfirmTransactionModal';

export function CreateDIDDockConfirmationModal({
  didName,
  didType,
  onCreateDID,
  visible,
  onClose,
}) {
  const didTypeMap = {
    diddock: 'did:dock',
    didkey: 'did:key',
  };
  return (
    <Modal
      visible={visible}
      onClose={onClose}
      onBackButtonPress={onClose}
      modalSize={0.8}>
      <VStack p={7}>
        <Typography variant="h1" mb={4}>
          {translate('confirm_transaction.title')}
        </Typography>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('didManagement.did_name')}
          </Typography>
          <Typography variant={'description'} mb={1}>
            {didName}
          </Typography>
        </NBox>

        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('didManagement.did_type')}
          </Typography>
          <Typography variant={'description'} mb={1}>
            {didTypeMap[didType]}
          </Typography>
        </NBox>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('confirm_transaction.fee')}
          </Typography>
          <AmountDetails amount={1} symbol={'DOCK'} />
        </NBox>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('confirm_transaction.total')}
          </Typography>
          <AmountDetails amount={1} symbol={'DOCK'} />
        </NBox>
        <LoadingButton
          mt={12}
          full
          {...addTestId('CreateNewDIDScreenDIDCreate')}
          onPress={onCreateDID}>
          {translate('confirm_transaction.title')}
        </LoadingButton>
      </VStack>
    </Modal>
  );
}
