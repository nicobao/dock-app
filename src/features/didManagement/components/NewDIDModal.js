import {Modal} from '../../../components/Modal';
import React from 'react';
import {Stack} from 'native-base';
import {
  DownloadIcon,
  OptionList,
  PlusCircleIcon,
  Theme,
  Typography,
} from '../../../design-system';
import {translate} from '../../../locales';
import {addTestId} from '../../../core/automation-utils';
import {navigate} from '../../../core/navigation';
import {Routes} from '../../../core/routes';

export function NewDIDModal({visible, onClose, onImportDID}) {
  const content = (
    <Stack p={8} testID="addDIDModal">
      <Typography variant="h1">{translate('didManagement.add_did')}</Typography>
      <OptionList
        mt={5}
        items={[
          {
            testID: addTestId('CreateNewDID').testID,
            title: translate('didManagement.create_new_did'),
            icon: (
              <PlusCircleIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: () => {
              onClose();
              navigate(Routes.DID_MANAGEMENT_NEW_DID);
            },
          },
          {
            testID: addTestId('ImportExistingDIDBtn').testID,
            title: translate('didManagement.import_existing_did'),
            icon: (
              <DownloadIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: onImportDID,
          },
        ]}
      />
    </Stack>
  );
  return (
    <Modal visible={visible} onClose={onClose} onBackButtonPress={onClose}>
      {content}
    </Modal>
  );
}
