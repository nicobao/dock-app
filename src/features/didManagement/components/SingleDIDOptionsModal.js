import {Stack} from 'native-base';
import React, {useCallback} from 'react';
import {
  DownloadIcon,
  BinIcon,
  OptionList,
  PencilIcon,
  Theme,
} from 'src/design-system';
import {translate} from 'src/locales';
import {Modal} from '../../../components/Modal';
import {Typography} from '../../../design-system';
import {addTestId} from '../../../core/automation-utils';
import {showConfirmationModal} from '../../../components/ConfirmationModal';
import {navigate} from '../../../core/navigation';
import {Routes} from '../../../core/routes';

export function SingleDIDOptionsModal({
  onClose,
  visible,
  didDocumentResolution,
  onDeleteDID,
}) {
  const showConfirmDeleteDIDModal = useCallback(() => {
    showConfirmationModal({
      type: 'warning',
      title: translate('didManagement.delete_did'),
      description: translate('didManagement.delete_description'),
      confirmText: translate('didManagement.delete'),
      cancelText: translate('didManagement.cancel'),
      onConfirm: () => {
        onDeleteDID(didDocumentResolution);
      },
      onCancel: () => {},
    });
  }, [didDocumentResolution, onDeleteDID]);

  const content = (
    <Stack p={8} testID="addAccountModal">
      <Typography variant="h1">
        {translate('didManagement.edit_did')}
      </Typography>
      <OptionList
        mt={5}
        items={[
          {
            testID: addTestId('EditDIDOption').testID,
            title: translate('didManagement.edit_did'),
            icon: (
              <PencilIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: () => {
              if (didDocumentResolution) {
                navigate(Routes.DID_MANAGEMENT_EDIT_DID, {
                  didDocumentResolution,
                });
              }
            },
          },
          {
            testID: addTestId('ExportDIDOption').testID,
            title: translate('didManagement.export_did'),
            icon: (
              <DownloadIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: () => {
              if (didDocumentResolution) {
                navigate(Routes.DID_MANAGEMENT_EXPORT_DID, {
                  didDocumentResolution,
                });
              }
            },
          },
          {
            testID: addTestId('DeleteDIDOption').testID,
            title: translate('didManagement.delete_did'),
            icon: (
              <BinIcon
                style={{
                  color: Theme.colors.secondaryIconColor,
                }}
              />
            ),
            onPress: () => {
              onClose();
              showConfirmDeleteDIDModal();
            },
          },
        ]}
      />
    </Stack>
  );
  return (
    <Modal visible={visible} onClose={onClose} onBackButtonPress={onClose}>
      {content}
    </Modal>
  );
}
