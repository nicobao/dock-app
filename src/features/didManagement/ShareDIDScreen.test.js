import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ShareDIDScreen} from './ShareDIDScreen';
import {PolkadotIcon} from '../../components/PolkadotIcon';

const mockStore = configureMockStore();

describe('ShareDIDScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const did = 'did:dock:5EfZQN8LKZvqwE9oda4wiognj3217bMj5AaDX4jBgEe3orJk';
    const wrapper = shallow(
      <ShareDIDScreen
        onCopyAddress={jest.fn()}
        did={did}
        didName={'TestDID'}
        accountIcon={<PolkadotIcon address={did} size={32} />}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
