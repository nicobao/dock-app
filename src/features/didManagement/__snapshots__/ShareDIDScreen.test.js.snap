// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareDIDScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="ShareDIDScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="ShareDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <BackButton
          accessibilityLabel="BackButton"
          testID="BackButton"
        />
      </Styled(View)>
      <Memo(ForwardRef(Stack))
        alignItems="center"
      >
        <Styled(Component)
          mb={2}
          variant="h1"
        >
          Share DID
        </Styled(Component)>
        <Styled(Component)
          variant="screen-description"
        >
          Share your Decentralized Identifier
        </Styled(Component)>
      </Memo(ForwardRef(Stack))>
      <Styled(ScrollView)
        marginLeft={26}
        marginRight={26}
      >
        <Memo(ForwardRef(Stack))
          alignItems="center"
          backgroundColor="#1F1F21"
          borderRadius={12}
          my={7}
          p={5}
        >
          <Memo(ForwardRef(Stack))
            bg="#fff"
            p={2}
          >
            <QRCode
              size={375}
              value="did:dock:5EfZQN8LKZvqwE9oda4wiognj3217bMj5AaDX4jBgEe3orJk"
            />
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            alignItems="center"
            direction="row"
            flex={1}
            pt={5}
          >
            <Memo(ForwardRef(Box))
              pr={2}
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/circle.png",
                  }
                }
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <Memo()
                address="did:dock:5EfZQN8LKZvqwE9oda4wiognj3217bMj5AaDX4jBgEe3orJk"
                size={32}
              />
            </Memo(ForwardRef(Box))>
            <Memo(ForwardRef(Stack))
              alignItems="flex-start"
              direction="column"
              flex={1}
            >
              <Styled(Component)
                variant="h3"
              >
                TestDID
              </Styled(Component)>
              <Styled(Component)
                numberOfLines={1}
              >
                did:dock:5EfZQN8LKZvqwE9oda4wiognj3217bMj5AaDX4jBgEe3orJk
              </Styled(Component)>
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(Stack))>
      </Styled(ScrollView)>
      <Memo(ForwardRef(Box))
        mb={7}
        mx={7}
      >
        <Button
          onPress={[MockFunction]}
          size="sm"
          startIcon={
            <Memo
              as={[Function]}
              name="copy-outline"
              size="sm"
            />
          }
        >
          Copy DID
        </Button>
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
